### PR TITLE
feat: Add mapping details for repated variable

### LINKED ISSUE #3187: Mappings not shown in SSR catalogue
**What version of EMX2 are you using (see footer)**
v10.18.1

**Describe the bug**
Mappings to harmonized variables are not shown in the variables view

**To Reproduce**
1. Go to 'https://data-catalogue.molgeniscloud.org/catalogue/ssr-catalogue/ATHLETE'
2. Click on 'variables'
3. Search for variable '[adhd_wave_1](https://data-catalogue.molgeniscloud.org/catalogue/ssr-catalogue/ATHLETE/variables/adhd_wave_1-ATHLETE_CDM-outcome_ath-ATHLETE_CDM?keys={%22name%22:%22adhd_wave_1%22,%22resource%22:{%22id%22:%22ATHLETE_CDM%22},%22dataset%22:{%22name%22:%22outcome_ath%22,%22resource%22:{%22id%22:%22ATHLETE_CDM%22}}})' 
4. This variable is mapped by INMA, EDEN and NINFEA but no mappings are shown.

**Expected behavior**
Matrix and variable details shows cohorts that have mapped to the selected variable(s)

### PR DESCRIPTION
**Pull request by connoratrug**
- handle case where variable has repeats
- fix  harmonisation matrix partial mapped due to repeats case

note harmonisation status of a <variable, cohort> pair is computed in aggregated ( combine info from repeats ) or single variable (each (repeated) variable gets a status  )  mode.

Closes #3187

### CODE REVIEW

File: apps/nuxt3-ssr/utils/harmonization.ts [hunk 1]
```diff
@@ -49,14 +83,14 @@ const calcStatusForSingleVariable = (
...
-const calcStatusForRepeatingVariable = (
+const calcStatusForAggregatedRepeatingVariable = (
   variable: IRepeatingVariableWithMapping,
   cohort: { id: string }
 ): HarmonizationStatus => {
   const statusList = !variable.repeats
     ? []
     : variable.repeats.map((repeatedVariable) => {
-        const resourceMapping = variable.mappings?.find((mapping) => {
+        const resourceMapping = repeatedVariable.mappings?.find((mapping) => {
```
Comment (connoratrug) [inline apps/nuxt3-ssr/utils/harmonization.ts:93]: this change is the main bug fix for the overall harmonisation matrix ( maybe also port to old catalogue)

**Review by BrendaHijmans — CHANGES_REQUESTED** [review]
the harmonisation status should be taken from the 'Variable mappings.match' column. For instance see the data here: See data: https://preview-emx2-pr-3218.dev.molgenis.org/catalogue-demo/catalogue/#/variable-mappings?_col=source,sourceDataset,sourceVariables,target,targetDataset,targetVariable,match,status,syntax,comments&_page=1
In the UI this is translated to a 'partial' match, this is wrong.

**Comment by BrendaHijmans** [comment]
The repeats are not visible in a matrix. Should still be implemented.

**Comment by BrendaHijmans** [comment]
I have added test data to test mapping to repeated variable in LifeCycle: sbp_0, with 215 repeats. This variable has been mapped by ENVIRONAGE, GenR and G21. See the old harmonisation details here: https://data-catalogue.molgeniscloud.org/catalogue/catalogue/#/variable-explorer/dbp_0?model=LifeCycle_CDM&tab=detail&sourceCohort=ENVIRONAGE

**Comment by BrendaHijmans** [comment]
This can also be used as a test case: PELAGIE is not in list Networks.cohorts, so mapping from PELAGIE to adhd_eval should not be visible when searching in LifeCycle variables. It should be visible in the overall 'super' catalogue. 

In the test data, PELAGIE is in the network ATHLETE. The variables for LifeCycle that it maps to, should also be visible there.

File: apps/nuxt3-ssr/components/harmonization/HarmonizationListPerVariable.vue [hunk 2]
```diff
@@ -1,16 +1,51 @@
...
+
+const aggregatedHarmonizationStatus = computed(() => {
+  return props.cohortsWithMapping
+    .map((cwm) => {
+      const status = Array.isArray(cwm.status) ? cwm.status : [cwm.status];
+      return { cohort: cwm.cohort, status };
+    })
+    .map((statusPerCohort) => {
+      if (
+        statusPerCohort.status.includes("partial") ||
```
Comment (mswertz) [inline apps/nuxt3-ssr/components/harmonization/HarmonizationListPerVariable.vue:19]: this logic, this has been checked with Marije?

File: apps/nuxt3-ssr/components/harmonization/HarmonizationListPerVariable.vue [hunk 3]
```diff
@@ -36,7 +74,10 @@ defineProps<{
     <div>
       <ul>
         <li
-          v-for="{ cohort, status } in cohortsWithMapping.slice(20, 30)"
+          v-for="{ cohort, status } in aggregatedHarmonizationStatus.slice(
```
Comment (mswertz) [inline apps/nuxt3-ssr/components/harmonization/HarmonizationListPerVariable.vue:77]: I try to understand this slicing

File: apps/nuxt3-ssr/components/harmonization/VariableDetails.vue [hunk 4]
```diff
@@ -11,60 +13,110 @@ const props = defineProps<{
...
+  if (props.variable.repeats) {
+    const baseMapping = props.variable.mappings?.find(
+      (mapping) => mapping.sourceDataset.resource.id === activeCohortId.value
+    )?.description;
+    const repeatMappings = props.variable.repeats.map((repeat) => {
+      return repeat.mappings?.find(
+        (mapping) => mapping.sourceDataset.resource.id === activeCohortId.value
+      )?.description;
+    });
+    return [baseMapping, ...repeatMappings];
```
Comment (mswertz) [inline apps/nuxt3-ssr/components/harmonization/VariableDetails.vue:39]: nice

**Review by mswertz — APPROVED** [review]

File: apps/nuxt3-ssr/components/harmonization/HarmonizationListPerVariable.vue (hunk 2 above)
Comment (connoratrug) [reply apps/nuxt3-ssr/components/harmonization/HarmonizationListPerVariable.vue:19]: yes , she suggested to replace with grid , i created a issue for that

File: apps/nuxt3-ssr/components/harmonization/HarmonizationListPerVariable.vue (hunk 3 above)
Comment (connoratrug) [reply apps/nuxt3-ssr/components/harmonization/HarmonizationListPerVariable.vue:77]: place holder spil into columns , can be replaced by scroll container ( the slice is not part of this PR, and older then the scroll container )

File: apps/nuxt3-ssr/components/harmonization/VariableDetails.vue (hunk 4 above)
Comment (connoratrug) [reply apps/nuxt3-ssr/components/harmonization/VariableDetails.vue:39]: well , these is some discussion if the baseMapping should be in the list  , suggest to fix ( if needed ) in separate issue, this pr should really have been split into multiple as it's really a fix of fixes and features :S

**Pull request merged**